pkg/api/write.go

```diff
@@ -100,6 +89,54 @@ func Write(writer pgmodel.DBInserter, elector *util.Elector, metrics *Metrics) h
 	})
 }
 
+func loadWriteRequest(r *http.Request) (*prompb.WriteRequest, error, string) {
+	var req *prompb.WriteRequest
+
+	switch r.Header.Get("Content-Type") {
+	case "application/x-protobuf":
+		compressed, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			return nil, err, "Read error"
+		}
+
+		reqBuf, err := snappy.Decode(nil, compressed)
+		if err != nil {
+			return nil, err, "Snappy decode error"
+		}
+
+		req = pgmodel.NewWriteRequest()
+		if err := proto.Unmarshal(reqBuf, req); err != nil {
+			pgmodel.FinishWriteRequest(req)
+			return nil, err, "Protobuf unmarshal error"
+		}
+	case "application/json":
```

Comment:
we should def support snappy (optionally) in this path as well

Reply:
@cevian I'd recommend doing that in a separate PR if it takes any non-trivial time